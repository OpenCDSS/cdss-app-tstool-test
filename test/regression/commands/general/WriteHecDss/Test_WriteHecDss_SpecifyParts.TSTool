# Write monthly data to DSS
# Test setup...
StartLog(LogFile="Results/Test_WriteHecDss_SpecifyParts.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# RemoveFile(InputFile="Results\Test_WriteHecDss_SpecifyParts_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_SpecifyParts.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
TS Alias1 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Month.Fpart1",Description="Test data",SetStart="1950-01",SetEnd="1952-03",Units="MGD",PatternValues="5,10,12,13,75")
TS Alias2 = NewPatternTimeSeries(NewTSID="Apart2:Bpart2-with-dashes..Cpart2.Month.Fpart2",Description="Test data",SetStart="1950-01",SetEnd="1952-03",Units="MGD",PatternValues="20,5,30,4,3,2")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_SpecifyParts_out.dss",TSList=AllMatchingTSID,TSID="Alias1",Type=PER-AVER,A="NewPartA",B="NewPartB",F="NewPartF")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_SpecifyParts_out.dss",TSList=AllTS,Type=PER-AVER,A="NewPartA")
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_SpecifyParts_out.dss",InputFile="Test_WriteHecDss_SpecifyParts.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_SpecifyParts.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_SpecifyParts.dssutl.out",InputFile2="Results/Test_WriteHecDss_SpecifyParts.dssutl.out",WarnIfDifferent=True)
