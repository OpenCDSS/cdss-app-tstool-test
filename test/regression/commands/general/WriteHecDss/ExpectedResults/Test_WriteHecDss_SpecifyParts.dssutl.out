  
 /NEWPARTA/BPART1-WITH-DASHES/CPART1/01JAN1950/1MON/FPART1/
 Start: 31JAN1950 at 2400 hours;   End: 31MAR1952 at 2400 hours;  Number:   27
 Units: MGD        Type: PER-AVER
 31JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 31AUG1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 31MAR1951, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 31OCT1951, 2400;    10.00   12.00   13.00   75.00    5.00   10.00
 ----------------------------------------------------------------------------
  
 /NEWPARTA/BPART2-WITH-DASHES/CPART2/01JAN1950/1MON/FPART2/
 Start: 31JAN1950 at 2400 hours;   End: 31MAR1952 at 2400 hours;  Number:   27
 Units: MGD        Type: PER-AVER
 31JAN1950, 2400;    20.00    5.00   30.00    4.00    3.00    2.00   20.00
 31AUG1950, 2400;     5.00   30.00    4.00    3.00    2.00   20.00    5.00
 31MAR1951, 2400;    30.00    4.00    3.00    2.00   20.00    5.00   30.00
 31OCT1951, 2400;     4.00    3.00    2.00   20.00    5.00   30.00
 ----------------------------------------------------------------------------
  
 /NEWPARTA/NEWPARTB/CPART1/01JAN1950/1MON/NEWPARTF/
 Start: 31JAN1950 at 2400 hours;   End: 31MAR1952 at 2400 hours;  Number:   27
 Units: MGD        Type: PER-AVER
 31JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 31AUG1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 31MAR1951, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 31OCT1951, 2400;    10.00   12.00   13.00   75.00    5.00   10.00
 ----------------------------------------------------------------------------
  
